README.md

```diff
@@ -970,6 +972,52 @@ If you are using a feature flag with binary variants, the `StatusOverride` prope
 
 In the above example, the feature is enabled by the `AlwaysOn` filter. If the current user is in the calculated percentile range of 10 to 20, then the `On` variant is returned. Otherwise, the `Off` variant is returned and because `StatusOverride` is equal to `Disabled`, the feature will now be considered disabled.
 
+### Variant Service
+
+Dependency injection can be wired up with the variant feature flag. Different implementations of a service interface can be considered as variant services. Specific variant of an interface can be retrieved from the dependency injection container based on the allocated variant of a variant feature flag. This could be done by using `IVariantServiceProvider<TService>`.
+
+``` C#
+    IVariantServiceProvider<IAlgorithm> algorithmServiceProvider;
+    ...
+
+    IAlgorithm forecastAlgorithm = await algorithmServiceProvider.GetServiceAsync(cancellationToken); 
+```
+
+The `IVaraintServiceProvider<IAlgorithm>` will retrieve a specific implementation of `IAlgorithm` from the dependency injection container. It can be registered by calling `WithvariantService<TService>` on the `IFeatureManagementBuilder`.
+
+``` C#
+services.AddFeatureManagement() 
+        .WithVariantService<IAlgorithm>("ForecastAlgorithm");
+```
+
+With the call above, `IAlgorithm` will be wired up with the variant feature flag `ForecastAlgorithm`. The `GetServiceAsync` method of `IVariantService<IAlgorithm>` will retrieve the variant service of `IAlgorithm` which matches the name of allocated variant of the `ForecastAlgorithm` flag, from the dependency injection container.
+
+``` javascript
+{
+    "ForecastAlgorithm": {
+        "Variants": [ 
+            { 
+                "Name": "AlgorithmBeta" 
+            }
+        ] 
+    }
+}
+```
+
+**Note:** The `WithvariantService<TService>` method will not register any implementation of `TService`. You need to register variant serices as `TService` to allow `IVariantServiceProvider<TService>` to retrieve them from the dependency injection container.

```

Comment:
```suggestion
**Note:** The `WithVariantService<TService>` method will not register any implementation of `TService`. You need to register variant serices as `TService` to allow `IVariantServiceProvider<TService>` to retrieve them from the dependency injection container.

```

Reply:
I feel like we should include a snippet of registering the AlgorithmBeta class as defined in this note. Just

```csharp
services.AddSingleton<IAlgorithm, AlgorithmBeta>();
```

Reply:
User can register AlgorithemBeta in many ways:
``` C#
services.AddSingleton<IAlgorithm, AlgorithmBeta>()
services.AddSingleton<IAlgorithm>(sp => new AlgorithmBeta())
services.AddSingleton<IAlgorithm>(new AlgorithmBeta())
```
I don't want to leave an impression here that AlgorithmBeta has to be registered like this. My thought is that when the example application is ready, we can put a link here and say "For more details, please refer to this example."

Reply:
How about this:
Which extension method to use for registration depends on your needs. Here is an example.
``` C#
services.AddSingleton<IAlgorithm, AlgorithmBeta>();
```